src/main/scala/viper/server/frontends/lsp/ViperServerService.scala

```diff
@@ -0,0 +1,159 @@
+/**
+  * This Source Code Form is subject to the terms of the Mozilla Public
+  * License, v. 2.0. If a copy of the MPL was not distributed with this
+  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+  *
+  * Copyright (c) 2011-2020 ETH Zurich.
+  */
+
+package viper.server.frontends.lsp
+
+import java.util.concurrent.{CompletableFuture => CFuture}
+
+import akka.actor.{PoisonPill, Props}
+import akka.pattern.ask
+import akka.util.Timeout
+import viper.server.core.{ViperBackendConfig, ViperCache, ViperCoreServer}
+import viper.server.frontends.lsp.VerificationState.Stopped
+import viper.server.utility.AstGenerator
+import viper.server.utility.Helpers.getArgListFromArgString
+import viper.server.vsi.VerificationProtocol.StopVerification
+import viper.server.vsi.{VerJobId, VerificationServer}
+import viper.silver.ast.Program
+
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class ViperServerService(args: Array[String]) extends ViperCoreServer(args) with VerificationServer {
+
+  protected var timeout: Int = _
+
+  def isRunningServer: Boolean = isRunning
+
+  var is_ready = false
+
+  def setReady(backend: BackendProperties): Unit = {
+    Coordinator.backend = backend
+    start()
+    is_ready = true
+    val param = BackendReadyParams("Silicon", false, true)
+    Coordinator.client.notifyBackendReady(param)
+    Log.info("The backend is ready for verification")
+  }
+
+  def setStopping(): Unit = {
+    Log.debug("Set Stopping... ")
+    if(isRunning){
+      isRunning = false
+      val params = StateChangeParams(Stopped.id)
+      Coordinator.sendStateChangeNotification(params, None)
+    } else {
+      Log.debug("Server stopped")
+    }
+  }
+  def setStopped(): Unit = {
+    Log.debug("Set Stopped. ")
+    if(isRunning){
+      isRunning = false
+      val params = StateChangeParams(Stopped.id)
+      Coordinator.sendStateChangeNotification(params, None)
+    } else {
+      Log.debug("Server stopped")
+    }
+  }
+
+  def verify(command: String): VerJobId = {
+    Log.debug("Requesting ViperServer to start new job...")
+
+    val arg_list = getArgListFromArgString(command)
+    val file: String = arg_list.last
+    val arg_list_partial = arg_list.dropRight(1)
+
+    // Parse file
+    val astGen = new AstGenerator(_logger.get)
+    var ast_option: Option[Program] = None
+    try {
+      ast_option = astGen.generateViperAst(file)
```

Comment:
Shouldn't this use AstWorker? In general, the entire class seems a bit redundant to me

Reply:
AstWorker did not exist when LSP was created. The current LSP version needs to be adapted to the recent changes. For now, I just added a comment.